### PR TITLE
Dynamic connection problem

### PR DESCRIPTION
**Pull request by vnormatov**

### CODE REVIEW

File: unionFind.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    connect(a, b) {
+
+        if (this.isConnected(a, b)) return;
+
+        if (isNaN(this._elements[a])) {
+            this._elements[a] = a;
+            // this._elements[1]=1;
+        }
+
+        this._elements[b] = isNaN(this._elements[b]) ? b : this._elements[b];
```
Comment (pavlom0x) [inline unionFind.js]: it will be better if we check `a` and `b` at the same way

File: unionFind.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                }
+            }
+        }
+        return count;
+    }
+
+
+    component(a) {
+
+
```
Comment (pavlom0x) [inline unionFind.js]: trailing empty line

File: unionFind.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        let components = [];
+        for (let i = 0; i < this._elements.length; i++) {
+            if (!isNaN(this._elements[i])) {
+                if (isUnique(components, this._elements[i])) {
+                    components[count] = this._elements[i];
+                    count++;
+                }
+            }
+        }
+        return count;
```
Comment (pavlom0x) [inline unionFind.js]: you can return `components.length` and remove this variable

File: unionFind.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        }
+        return this._elements[a] === this._elements[b];
+    }
+
+    count() {
+        let count = 0;
+        let components = [];
+        for (let i = 0; i < this._elements.length; i++) {
+            if (!isNaN(this._elements[i])) {
+                if (isUnique(components, this._elements[i])) {
```
Comment (pavlom0x) [inline unionFind.js]: `isUnique` can be replaced with `indexOf`

File: test/unioonFind.spec.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    it('should #component return component name', function () {
+        let uf = new algolib.UnionFind();
+        uf.connect(1, 2);
+        uf.connect(2, 3);
+        uf.connect(1, 5);
+        let nameOfComponent = uf.component(1);
+        nameOfComponent.should.be.eql(5);
+    })
+
+    it('should #component return component name', function () {
```
Comment (pavlom0x) [inline test/unioonFind.spec.js]: test description should be unique

File: test/unioonFind.spec.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    })
+
+    it('should #count return number of components if we have 1 component', function () {
+        let uf = new algolib.UnionFind();
+        uf.connect(3, 6);
+        let numberOfComponents = uf.count();
+        numberOfComponents.should.be.eql(1);
+    })
+
+    it('should #count return number of components if we have 3 components', function () {
```
Comment (pavlom0x) [inline test/unioonFind.spec.js]: #count should return number not only when we have 3 components

**Review by pavlom0x — CHANGES_REQUESTED** [review]
plz remove redundant empty line

File: unionFind.js (hunk 1 above)
Comment (vnormatov) [reply unionFind.js]: ok, I did it for me to better remember the syntax)

File: unionFind.js (hunk 3 above)
Comment (vnormatov) [reply unionFind.js]: fixed

File: unionFind.js (hunk 4 above)
Comment (vnormatov) [reply unionFind.js]: fixed

File: unionFind.js (hunk 2 above)
Comment (vnormatov) [reply unionFind.js]: ok)

File: test/unioonFind.spec.js (hunk 5 above)
Comment (vnormatov) [reply test/unioonFind.spec.js]: fixed

File: test/unioonFind.spec.js (hunk 6 above)
Comment (vnormatov) [reply test/unioonFind.spec.js]: ok)

**Review by pavlom0x — APPROVED** [review]
good work

**Comment by pavlom0x** [comment]
if nothing to add than merge it

**Pull request merged**